R/acuity.R

```diff
@@ -0,0 +1,96 @@
+acuityview <- function(image, obj_dist, obj_width, eye_res){
+  
+  # Images must currently be square, with dimensions a power of 2
+  # TODO: kill this limitation
+  if(dim(image)[1] != dim(image)[2] || !is.element(dim(image)[1], 2^c(1:100))){
+    stop("Image must be square, and with dimensions a power of 2")
+  }
+  
+  # Image width in degrees
+  width_deg <- 57.2958 * (2 * atan(obj_width / obj_dist / 2))
+  
+  # Image width in pixels
+  width_pix <- dim(image)[2]
+  
+  # Image center
+  center <- round(width_pix / 2) + 1
+  
+  # Create a blur matrix with dimensions equal to the image
+  blur <- matrix(NA, nrow = width_pix, ncol = width_pix)
+  for (i in 1:width_pix){
+    for (j in 1:width_pix) {
+      x <- i - center
+      y <- j - center
+      freq <- round(sqrt(x^2 + y^2)) / width_pix * (width_pix / width_deg)
+      mySin <- y / sqrt(x^2 + y^2)
+      myCos <- x / sqrt(x^2 + y^2)
+      eye_res2 <- eye_res * eye_res /sqrt((eye_res * myCos)^2 + (eye_res * mySin)^2)
+      blur[i,j] <- exp(-3.56 * (eye_res2 * freq)^2)
+    }
+  }
+  
+  # Force the center pixel to value 1
+  blur[center, center] = 1
+  blur <<- blur
```

Comment:
I think CRAN is really reluctant to accept changes with `<<-`. Can you explain why we need this here? We don't have any other choice?